src/metrics_regression.rs

```diff
@@ -69,6 +69,16 @@ pub trait SingleTargetRegression<F: Float, T: AsSingleTargets<Elem = F>>:
         }
     }
 
+       /// Mean absolute percentage error between two continuous variables
+       fn mean_absolute_percentage_error(&self, compare_to: &T) -> Result<F> {
+        self.as_single_targets()
+            .sub(&compare_to.as_single_targets())
+            .mapv(|x| x.abs())
+            .div(&self.as_single_targets().mapv(|x| x.abs().max(F::min_positive_value())))
```

Comment:
You can call `mapv(|x| x.abs())` after the `div` call and remove the `mapv` call inside the `div`, since you only need to call `abs` once. The `.max(F::min_positive_value())` part isn't necessary since you'll end up with `INF` either way if `self` is 0.

Reply:
> You can call mapv(|x| x.abs()) after the div call and remove the mapv call inside the div, since you only need to call abs once.
Totally! Good call
>The .max(F::min_positive_value()) part isn't necessary since you'll end up with INF either way if self is 0.
Not exactly. Using `.max(F::min_positive_value())`results in very large number and not `inf`; conversely dividing by `0` results in `inf` - you can try it out in tests

Reply:
IMO if `self` is 0 then the output should be `INF`, since you're literally dividing by 0

Reply:
Changed. Allowing division by 0 and calling `mapv(|x| x.abs())` after `div`  and removed inner `mapv`